legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java

```diff
@@ -387,10 +391,29 @@ private Map<String, String> extractPlaceHolderKeyValues(Datasets datasets, Execu
     {
         Map<String, String> placeHolderKeyValues = new HashMap<>();
         Optional<Long> nextBatchId = getNextBatchId(datasets, executor, transformer, ingestMode);
+        Optional<Map<OptimizationFilter, List<Object>>> optimizationFilters = getOptimizationFilterBounds(datasets, executor, transformer, ingestMode);
         if (nextBatchId.isPresent())
         {
             placeHolderKeyValues.put(BATCH_ID_PATTERN, nextBatchId.get().toString());
         }
+        if (optimizationFilters.isPresent())
+        {
+            for (OptimizationFilter filter : optimizationFilters.get().keySet())
+            {
+                Object lowerBound = optimizationFilters.get().get(filter).get(0);
+                Object upperBound = optimizationFilters.get().get(filter).get(1);
+                if (lowerBound instanceof Date)
```

Comment:
I was trying to fin a test case for this, and not sure I saw one.  I was hoping to get more details into what these quotes do, and why just Date does not required quoting.  Can you point me to a test or add one?  Also explain the rational of the if/else?

Reply:
The test that I added in UnitemporalDeltaTest.java in h2 module tests both the if branch and else branch. In the test there are two optimization filters, "expiryDate" of date type and "id" of integer type.

The rationale of having this block is as follows:

Firstly, we always produce the placeholders text surrounded by single quotes, example below:
`(sink."column" >= **'{ID_LOWER_BOUND}'**) AND (sink."column" <= **'{ID_UPPER_BOUND}'**)`

When we replace the placeholders with actual values, we want to do different things with different types (we only support date type or number type for optimization filters):

- For date type, we want to retain the single quotes, such that after replacement it will look like this: `(sink."column" >= '2023-01-01') AND (sink."column" <= '2023-01-02')`
- For number type, we do not want to retain the single quotes, such that after replacement it will look like this: `(sink."column" >= 1) AND (sink."column" <= 2)`

Reply:
But the code tells a different story.  If you have a Date (look at the if condition `if (lowerBound instanceof Date)`), don't add quotes, hence why I was confuse, and trying to understand the rational for this change.  

Do Im missing something or is the code buggy and we should look into it?  I suspect a better instacne of will be `if (lowerbound instanceof Number)` so that non-numbers keep the quotes.

Reply:
Think the code is okay. I think the confusion is `placeHolderKeyValues` appears to be a map where the key is the 'find' term and the value is the 'replace' term.

So for the if clause (Date), the pattern to search for is just `{LOWER_BOUND}` with no quotes, so the quotes are kept upon replacement.

Whereas for the else clause, the pattern to search for is `'{LOWER_BOUND}'` with quotes, so the quotes will be removed upon replacement.

Test does look like it covers this.

Reply:
Separate question though: should we switch from `else` to something like `else if (lowerBound instanceof Number)` and add a separate `else` clause that throws?

Reply:
These are the supported data types for lower bound and Upper Bound values:
INT, INTEGER, INT64, BIGINT, FLOAT, DOUBLE, DECIMAL, DATE
Out of them , only DATE needs a quote and other data types do not need a quote.
That's why it was written with if and a common else block.

I think we can change this to switch case to be more cleaner.

Reply:
> Separate question though: should we switch from `else` to something like `else if (lowerBound instanceof Number)` and add a separate `else` clause that throws?

Sure, fixed this!

Reply:
Thanks for the explanation!  This make sense now, but I feel still too convoluted :) 

While not a feedback item to address now, we should change the template query to avoid having the quotes there, and let the value we want to replace with come with the quotes, if needed.

For example:  `where COL >= {LOWER_BOUND}`

then the replacement pair will provide the quotes to the value, rather than the other way around:

```java
placeHolderKeyValues.put(filter.lowerBoundPattern(), SINGLE_QUOTE + lowerBound.toString() + SINGLE_QUOTE);
```